src/main/webapp/assets/js/feedScript.js

```diff
@@ -628,32 +825,48 @@ async function submitModal() {
     if (!latLngResult && modalLocation !== cachedModalAddress) {
       cachedModalAddress = modalLocation;
       const modalError = document.createElement('div');
-      modalError.setAttribute('id', 'modal-error');
+      modalError.setAttribute('id', 'modal-map-error');
       modalError.innerText =
         `We couldn't find address '${modalLocation}'. ` +
         'Please check your address for errors. ' +
         'If you wish to submit anyway, no pin will be added to the map.';
-      document.getElementById('modal-form')
-          .insertBefore(modalError, document.getElementById('modal-submit'));
+      modalForm.insertBefore(modalError, document.getElementById('modal-submit'));
+      submitModalButton.disabled = false;
     // Else, if the invalid address is the same as we last checked
     // or the address is just plain valid, then add the post to the Datastore.
     } else {
-      const modalError = document.getElementById('modal-error');
+      const modalError = document.getElementById('modal-map-error');
       if (modalError) {
-        document.getElementById('modal-form').removeChild(modalError);
+        modalForm.removeChild(modalError);
       }
 
       // (0,0) denotes a nonexistent lat/long, since it's a location in the ocean + is falsy.
       const lat = latLngResult ? latLngResult.lat : 0;
       const lng = latLngResult ? latLngResult.long : 0;
       url = `/postData?collegeId=${collegeId}&lat=${lat}&lng=${lng}`;
       modalForm.action = url;
-      modalForm.submit();
-      createPostButton.focus();
+
+      // Show submitted box.
+      modalCard.style.display = 'none';
+      modalSubmitted.style.display = 'block';
+      modalSubmitted.focus();
+      setTimeout(function() {
```

Comment:
Could

Reply:
It closes the form modal and shows the submit message! Added a comment in the code to increase readability.